### PR TITLE
Adjust forecast solar data time offset to match

### PR DESCRIPTION
**Pull request by arthurbenemann**
## Breaking change
No
## Proposed change
Current forecast data appears time shifted to solar production graph.  `forecast_solar` integration appears to generate correct data reported as hourly reset generation, at the end-of-hour. This matches well to actual hourly production values, but on the front-end card it appears shifted by 1h.

This correction is  removing a 1h delay (3.6e6 ms) so the plots align on the frontend. Not sure if this is the best location for the fix, as it will also add to the weekly/montly forecast (but in that case 1h delay is minimal on the graph).

Current output (frontend 20220907.2 - latest)
image

With the PR applied:
image

Diagnostics of solar forecast integration: 
[config_entry-forecast_solar-5bd412b4d71d1d33e198cbb1f43e3ab6.json (4).txt](https://github.com/home-assistant/frontend/files/9639753/config_entry-forecast_solar-5bd412b4d71d1d33e198cbb1f43e3ab6.json.4.txt)

Relevant links:

- https://www.home-assistant.io/integrations/forecast_solar 
- https://community.home-assistant.io/t/cant-get-forecast-solar-forecast-to-line-up-with-my-production/432205

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```
Requires solar to be setup in the energy dashboard, with forecast_solar also setup to generate forecast data. 
A live system makes it easier to correlate the data
```

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/core/issues/74375
- This PR is related to issue: potentially related to https://github.com/home-assistant/core/pull/77217
- This PR is related to issue or discussion: https://community.home-assistant.io/t/cant-get-forecast-solar-forecast-to-line-up-with-my-production/432205

- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally. 
- **Note tested only via chromium debugger, I have not found a good way to run a separate front-end for testing on my live system that contains usable data**
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @arthurbenemann

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant/frontend#13855).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by arthurbenemann** [comment]
Updated main PR post with screen-captures

File: src/panels/lovelace/cards/energy/hui-energy-solar-graph-card.ts [hunk 1]
```diff
@@ -455,7 +455,7 @@ export class HuiEnergySolarGraphCard
               } else {
                 dateObj.setMinutes(0, 0, 0);
               }
-              const time = dateObj.getTime();
+              const time = dateObj.getTime()-3.6e6;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/energy/hui-energy-solar-graph-card.ts:458]: We should do the adjustment on line 446, so we calculate with the correct time from the start

**Comment by binarymelon** [comment]
@arthurbenemann just wondering if this is going to get updated?

**Comment by bramkragten** [comment]
Stale, feel free to re-open when you plan to continue working on this.

**Pull request closed without merging**